client/src/partials/vouchers/complex.js

```diff
@@ -255,8 +270,85 @@ function ComplexJournalVoucherController(Vouchers, $translate, Accounts, Currenc
     // init sum debit and credit
     vm.sumDebit  = 0;
     vm.sumCredit = 0;
+
+    // init financial accounts
+    vm.financialAccount = [];
+
+    Cashbox.read(null, { detailed: 1 })
+    .then(function (list) {
+
+      list.forEach(function (item) {
+
+        if (vm.financialAccount.indexOf(item.account_id) === -1) {
+          vm.financialAccount.push(item.account_id);
+        }
+
+        if (vm.financialAccount.indexOf(item.transfer_account_id) === -1) {
+          vm.financialAccount.push(item.transfer_account_id);
+        }
+
+      });
+
+    })
+    .catch(Notify.handleError);
   }
 
+  /** check use of financial accounts */
+  function isFinancial() {
+
+    vm.financialTransaction = false;
+    for (var i = 0; i < vm.rows.length; i++) {
+      if (vm.rows[i].account && vm.financialAccount.indexOf(vm.rows[i].account.id) !== -1) {
+        vm.financialTransaction = true;
+        break;
+      }
+    }
+
+    // prevent persistent value
+    vm.voucher.type_id = vm.financialTransaction ? vm.voucher.type_id : null;
+
+  }
+
+  /* ============================= Transfer Type ============================= */
+  vm.transferType = Vouchers.transferType;
+
+  vm.buildDescription = buildDescription;
+
+  function groupType() {
+    vm.incomes = vm.transferType.filter(function (item) {
+      return item.incomeExpense === 'income';
+    });
+    vm.expenses = vm.transferType.filter(function (item) {
+      return item.incomeExpense === 'expense';
+    });
+  }
+
+  function buildDescription() {
+    var type,
+        current = new Date(),
+        description = String(Session.project.abbr).concat('/VOUCHER');
+
+    if (vm.voucher.type_id) {
+
+      type = JSON.parse(vm.voucher.type_id);
+
+      vm.incomeExpense = type.incomeExpense;
+
+      vm.descriptionPrefix = description
+        .concat('/', type.prefix)
+        .concat('/', current.toDateString());
```

Comment:
For this type of thing, it would be better to use a `moment` date, formatted as 'DD-MM-YYYY'.  It makes it easier to read.